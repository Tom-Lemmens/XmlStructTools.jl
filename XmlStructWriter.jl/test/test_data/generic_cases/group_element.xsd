<?xml version="1.0"?>
<schema xmlns="http://www.w3.org/2001/XMLSchema" xmlns:TestGroup="TestGroup" targetNamespace="TestGroup">

    <element name="document" type="TestGroup:documentType"/>

    <group name="TestGroup1">
        <annotation>
            <documentation>An example of an xsd group.</documentation>
        </annotation>
        <sequence>
            <element name="Element_string" type="string"/>
            <element name="Element_double" type="double"/>
            <element name="Element_boolean" type="boolean"/>
        </sequence>
    </group>

    <group name="TestGroup2">
        <annotation>
            <documentation>An example of an xsd group.</documentation>
        </annotation>
        <sequence>
            <element name="Element_string_2" type="string"/>
            <element name="Element_double_2" type="double"/>
            <element name="Element_boolean_2" type="boolean"/>
        </sequence>
    </group>

    <complexType name="TestComplexType1">
        <annotation>
            <documentation>Example of a complex XSD type that uses a group.</documentation>
        </annotation>
        <group ref="TestGroup:TestGroup1"></group>
    </complexType>

    <complexType name="TestComplexType2">
        <annotation>
            <documentation>Example of a complex XSD type that uses a group and an element.</documentation>
        </annotation>
        <sequence>
            <group ref="TestGroup:TestGroup1"></group>
            <element name="Element_boolean_second" type="boolean"/>
            <group ref="TestGroup:TestGroup2"></group>
        </sequence>
    </complexType>

    <complexType name="TestComplexType3">
        <annotation>
            <documentation>Example of a complex XSD type that uses a nested group.</documentation>
        </annotation>
        <group ref="TestGroup:TestGroup3"></group>
    </complexType>

    <group name="TestGroup3">
        <annotation>
            <documentation>An example of an xsd group.</documentation>
        </annotation>
        <sequence>
            <group ref="TestGroup:TestGroup1"/>
            <element name="Element_boolean2" type="boolean"/>
        </sequence>
    </group>

    <complexType name="documentType">
        <sequence>
            <element name="TestElement1" type="TestGroup:TestComplexType1"/>
            <element name="TestElement2" type="TestGroup:TestComplexType2"/>
            <element name="TestElement3" type="TestGroup:TestComplexType3"/>
        </sequence>
    </complexType>

</schema>