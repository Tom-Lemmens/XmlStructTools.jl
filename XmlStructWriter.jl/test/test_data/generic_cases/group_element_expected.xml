<?xml version="1.0" encoding="utf-8"?>

<TestGroup:document xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="TestGroup group_element.xsd" xmlns:TestGroup="TestGroup">
    <TestElement1>
        <Element_string>wwww</Element_string>
        <Element_double>52.65</Element_double>
        <Element_boolean>true</Element_boolean>
    </TestElement1>
    <TestElement2>
        <Element_string>wwww</Element_string>
        <Element_double>52.65</Element_double>
        <Element_boolean>true</Element_boolean>
        <Element_boolean_second>false</Element_boolean_second>
        <Element_string_2>uuiu</Element_string_2>
        <Element_double_2>10</Element_double_2>
        <Element_boolean_2>true</Element_boolean_2>
    </TestElement2>
    <TestElement3>
        <Element_string>ttt</Element_string>
        <Element_double>77.99</Element_double>
        <Element_boolean>false</Element_boolean>
        <Element_boolean2>true</Element_boolean2>
    </TestElement3>
</TestGroup:document>